spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/query/GqlDatastoreQuery.java

```diff
@@ -112,26 +113,79 @@ public GqlDatastoreQuery(Class<T> type, DatastoreQueryMethod queryMethod,
 	@Override
 	public Object execute(Object[] parameters) {
 		GqlQuery query = bindArgsToGqlQuery(this.gql, getParamTags(), parameters);
-		Iterable found = this.datastoreTemplate.queryKeysOrEntities(query, this.entityType);
+		boolean returnTypeIsCollection = this.queryMethod.isCollectionQuery();
+		Class returnedItemType = this.queryMethod.getReturnedObjectType();
+
+		boolean isNonEntityReturnType = isNonEntityReturnedType(returnedItemType);
+
+		Iterable found = isNonEntityReturnType
+				? this.datastoreTemplate.query(query, x -> {
+					Object mappedResult;
```

Comment:
can this be moved to a separate static method instead of lambda?

Reply:
@dmitry-s both changes made!